integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexJobs.scala

```diff
@@ -55,14 +57,27 @@ class DistributedIndexJob extends AbstractIndexJob {
     val (response, time) = logTime {
       try {
         val spark = SparkSQLUtil.getSparkSession
-        indexFormat.setTaskGroupId(SparkSQLUtil.getTaskGroupId(spark))
-        indexFormat.setTaskGroupDesc(SparkSQLUtil.getTaskGroupDesc(spark))
+        // In case of presto with index server flow, sparksession will be null
+        if (null != spark) {
```

Comment:
is it possible to add a configuration to CarbonProperties that can tell us whether its a presto flow or not. Property should not be exposed to user. just for internal purpose.

This null check looks dirty.

Reply:
handled